File: Harpia-Health-Analysis-WS/src/main/java/com/harpia/HarpiaHealthAnalysisWS/model/timer/PatientTimer.java
package com.harpia.HarpiaHealthAnalysisWS.model.timer;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "patient_timer", uniqueConstraints = @UniqueConstraint(columnNames = "patient_id"))
public class PatientTimer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column
    private long id;
    @Column
    private int hours;
    @Column
    private int minutes;
    @Column(name = "patient_id")
    @NonNull
    private long patientId;

    @Override
    public String toString() {
        return "PatientTimer{" +
                "id=" + id +
                ", hours=" + hours +
                ", minutes=" + minutes +
                ", patientId=" + patientId +
                '}';
    }

}

File: Flutter-Harpia-Health-Analysis/lib/model/specialitem/doctor/PatientTimer.dart
class PatientTimer {
  int _hours = 0;
  int _minutes = 0;
  int _patientId = -1;

  int get hours => _hours;

  set hours(int value) {
    _hours = value;
  }

  int get minutes => _minutes;

  set minutes(int value) {
    _minutes = value;
  }

  int get patientId => _patientId;

  set patientId(int value) {
    _patientId = value;
  }

  @override
  String toString() {
    return 'PatientTimer{_hour: $_hours, _min: $_minutes, _patientId: $_patientId}';
  }
}
